Model/Data/AmwalOrderDetails.php

```diff
@@ -118,7 +146,10 @@ private function isPayValid($amwalOrderId)
         }
         return $orderState === 'pending_payment';
     }
-
+    /**
+     * @param $order
```

Comment:
I think we can add OrderInterface as a typehint here and in the function parameter